projects/bp-gallery/src/components/common/DonateButton.tsx

```diff
@@ -0,0 +1,122 @@
+import { Button, Dialog } from '@mui/material';
+import { PayPalButtons, PayPalScriptProvider } from '@paypal/react-paypal-js';
+import { useEffect, useRef, useState } from 'react';
+import CurrencyInput from 'react-currency-input-field';
+import './DonateButton.scss';
+import PrimaryButton from './PrimaryButton';
+import { OnApproveActions, OnApproveData } from '@paypal/paypal-js';
+import { useTranslation } from 'react-i18next';
+
+const DonateButton = ({
+  clientId,
+  donationText,
+  purposeText,
+}: {
+  clientId: string;
+  donationText?: string;
+  purposeText?: string;
+}) => {
+  const { t } = useTranslation();
+  const [isClicked, setIsClicked] = useState(false);
+  const [isOpen, setIsOpen] = useState(false);
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [dialogText, setDialogText] = useState('');
+  const [donation, setDonation] = useState('0.00');
+  const transitionRef = useRef(null);
+  const paypalOptions = {
+    'client-id': clientId,
+    currency: 'EUR',
+  };
+
+  const onApprove = (data: OnApproveData, actions: OnApproveActions) => {
+    return actions.order.capture().then(() => {
```

Comment:
I actually get an error saying `'actions.order' is possibly 'undefined'` here. Don't you get that, too?

Reply:
Yes, but if we did that this function would return Promise|undefined and onApprove needs a function that returns a Promise. Furthermore the approve function wouldn't be called if action.order wasn't defined

Reply:
I still don't think it is a good idea to leave TypeScript errors in the code. Even if you are sure that `action.order` will never be `undefined`, you can still check for it, that shouldn't hurt. As for your `Promise<...> | undefined` problem, you can use an `async` function instead:
```ts
const onApprove = async (data: OnApproveData, actions: OnApproveActions) => {
  await actions.order?.capture();
  setDialogText(t('archives.edit.paypal.thankyou-label', { amount: donation }));
  setIsDialogOpen(true);
};
```
This has the nice benefit of also removing the `.then` and the accompanying nesting ;)